addons/binding/org.openhab.binding.tivo/README.md

```diff
@@ -0,0 +1,235 @@
+![Tesla](doc/TiVo_lockup_BLK.png)
+
+## TiVo Binding
+This binding integrates [TiVo](https://www.tivo.com/) Digital Video Recorders (DVR) that support the Tivo [TiVo TCP Control Protocol v1.1](https://www.tivo.com/assets/images/abouttivo/resources/downloads/brochures/TiVo_TCP_Network_Remote_Control_Protocol.pdf).
+
+## Supported Things
+Most TiVo DVRs that support network remote control can be managed/supported by this binding.  Check the web site of your service provider for the precise specification of the TiVo box they have provided.
+
+All TiVo devices must:
+
+ 1. be connected to a local area TCP/IP network that can be reached by the openHAB instance (this is not the WAN network interface used by cable service providers to provide the TV signals).    
+ 2. have the Network Remote Control function enabled to support discovery and control of the device.  This setting can be found using the remote control at:
+
+    * Tivo branded boxes - using the remote go to TiVo Central > Messages & Settings > Settings > Remote, CableCARD & Devices > Network Remote Control.  Choose Enabled, press Select.
+    * Virgin Media branded boxes - using the remote select Home > Help and Settings > Settings > Devices > Network Remote Control.  Select the option Allow network based remote controls.
+
+## Binding Configuration
+The binding requires no manual configuration.  Tivo devices with the network remote control interface enabled, will be displayed within the Inbox.  
+
+You can also add these manually, you will need to specify the LAN IP address of your Tivo Device.
+
+## Thing Configuration
+
+Auto-discovery is recommended for the discovery and creation of TiVo things, however they can also be created using the .things file format.  The following minimum parameters should be used:
+
+```
+Thing tivo:sckt:test_device[deviceName = "Test Device", address="192.168.0.19"]
+```
+
+Where:
+* **test_device** is the unique thing ID for the device (alpha numeric, no spaces)
+* **device name** is the name of the device (if omitted the name of the device will be specified as 'My Tivo') 
+* **address** the IP address or host name of the device
+
+See the Parameters section below, for the definition of the other optional parameter field names / values.
+
+## Channels
+
+All devices support the following channels (non exhaustive):
+
+| Channel Type ID | Item Type    | Display Name | Description  |
```

Comment:
You would make me happy by formatting the tables in the MD file as well :-D

http://www.tablesgenerator.com/markdown_tables

Select 'File' > 'Past table data ...' and past the unformatted mark down table, you can then copy paste the formatted table 👍